### ISSUE TITLE
Cannot get the merge alias of a map

### ISSUE THREAD

**Issue by mikefarah** [mikefarah/yq#1942]
### Discussed in https://github.com/mikefarah/yq/discussions/1937

<div type='discussions-op-text'>

<sup>Originally posted by **qaisjp** February  6, 2024</sup>
I have the following YAML:

```yaml
foo: &foo
  hello: world
bar:
  <<: *foo
```

And would like to compute the following values based on the "parent" of each object

```yaml
foo: &foo
  parent: null
bar:
  parent: foo   # bar inherits from "foo", so the value of bar.parent should be "foo"
```

I've tried the following filters, but can't seem to figure it out!

## filter attempt 1

the following queries:

```jq
with_entries(.value |= {
  "parent": . | alias
})
```
```jq
with_entries(.value |= {
  "parent": alias
})
```

produces

```yaml
foo: &foo
  parent: ""
bar:
  parent: ""  # expected "foo"
```

## filter attempt 2

```jq
with_entries(.value |= {
  "parent": .["<<"]
})
```

produces

```yaml
foo: &foo
  parent: null
bar:
  parent: null
```

## filter attempt 3

```jq
with_entries(.value |= {
  "parent": (.["<<"] | alias)
})
```

produces

```yaml
foo: &foo
  parent: "null"
bar:
  parent: "null"
```

## Note that it works fine with  _anchors_

```yq
with_entries(.value |= {
  "anchor_name": . | anchor
})
```

produces

```yaml
foo: &foo
  anchor_name: foo
bar:
  anchor_name: ""
```

## `<<: *` is parsed correctly

When I just pipe my yaml through with the `.` filter, I get

```yaml
foo: &foo
  hello: world
bar:
  !!merge <<: *foo
```

Which is correct — according to the docs it says:

> `yq` supports merge aliases (like `<<: *blah`) however this is no longer in the standard yaml spec (1.2) and so `yq` will automatically add the `!!merge` tag to these nodes as it is effectively a custom tag.

(also mentioned at https://github.com/mikefarah/yq/issues/680#issuecomment-764542837)

but why can't I read the merge alias at `.["<<"]`?

---

Any pointers? 🙁  Thanks!</div>

**Comment by mikefarah** [comment]
Will fix in the next release :)

**Comment by mikefarah** [comment]
Fwiw - when fixed you can get the alias name by:

```
yq '.bar["<<"] | alias' file.yaml
```

**Comment by mikefarah** [comment]
Fixed in 4.41.1

**Comment by qaisjp** [comment]
Thanks!
